### PR TITLE
[NPU] Clean `skip_tests_config`

### PR DESCRIPTION
**Pull request by MirceaDan99**
- *item1*
 - *...*

### Tickets:
 - *E-120273*

### CODE REVIEW

File: src/plugins/intel_npu/tests/functional/shared_tests_instances/skip_tests_config.cpp [hunk 1]
```diff
@@ -4,969 +4,9 @@
...
-        return _skipRegistry;
-    }();
-    // clang-format on
-
-    std::vector<std::string> matchingPatterns;
-    const auto currentTestName = getCurrentTestName();
-    matchingPatterns.emplace_back(skipRegistry.getMatchingPattern(currentTestName));
-
-    return matchingPatterns;
+    return std::vector<std::string>{};
```
Comment (ArtemySkrebkov) [inline src/plugins/intel_npu/tests/functional/shared_tests_instances/skip_tests_config.cpp:11]: Do we expect no disabled tests?

It is just not very clear what is the motivation for the change.

Comment (sbutnari) [reply src/plugins/intel_npu/tests/functional/shared_tests_instances/skip_tests_config.cpp:11]: We expect disabled tests, the motivation here is primarily to do a general cleanup. To try and declutter the skip_configs while also migrating tests from the old repo.
